### PR TITLE
Add quilt patch refresh to debian changelog

### PR DESCRIPTION
**Pull request by TheRealFalcon**
Example of most recent log generated:
```
cloud-init (23.4.1-0ubuntu1~22.04.3) UNRELEASED; urgency=medium

  * d/cloud-init.maintscript: remove /etc/cloud/clean.d/README
  * d/cloud-init.logrotate: add logrotate config for cloud-init
  * refresh patches:
    - d/p/retain-netplan-world-readable.patch
    - d/p/status-do-not-remove-duplicated-data.patch
    - d/p/status-retain-recoverable-error-exit-code.patch
  * Upstream snapshot based on upstream/main at 33148a07.
```

### CODE REVIEW

File: scripts/new_upstream_snapshot.py [hunk 1]
```diff
@@ -349,6 +349,9 @@ def refresh_patches(commitish) -> bool:
             f"patches: \n" + "\n".join(patches)
         )
         sh(f"git commit --no-verify -m '{commit_msg}' {' '.join(patches)}")
+        add_msg_to_changelog(
+            "  * refresh patches:\n    - " + "\n    - ".join(patches)
```
Comment (blackboxsw) [inline scripts/new_upstream_snapshot.py:353]: This makes for a lot of duplicated full file path names.
```
 * refresh patches
    - debian/patches/X.patch
    - debian/patches/Y.patch
    - debian/patches/Z.patch
```
How about we reduce this to 
```
 * refresh debian/patches:
    - X.patch
    - Y.patch
    - Z.patch
```

```diff
--- a/scripts/new_upstream_snapshot.py
+++ b/scripts/new_upstream_snapshot.py
@@ -14,6 +14,7 @@ The commitish defaults to 'main'
 """
 
 import argparse
+import os
 import re
 import subprocess
 import sys
@@ -350,7 +351,8 @@ def refresh_patches(commitish) -> bool:
         )
         sh(f"git commit --no-verify -m '{commit_msg}' {' '.join(patches)}")
         add_msg_to_changelog(
-            "  * refresh patches:\n    - " + "\n    - ".join(patches)
+            "  * refresh debian/patches:\n    - " +
+            "\n    - ".join([os.path.basename(p) for p in patches])
         )
         return True
     else:
```

**Review by blackboxsw — APPROVED** [review]
bikeshed on debian/changelog content. +1 either way.

File: scripts/new_upstream_snapshot.py (hunk 1 above)
Comment (TheRealFalcon) [reply scripts/new_upstream_snapshot.py]: Gahhh, sorry. I had actually already changed this locally and rebased it, but never pushed it. I did a force push for it.

File: scripts/new_upstream_snapshot.py [hunk 2]
```diff
@@ -339,20 +339,24 @@ def refresh_patches(commitish) -> bool:
...
+    if not patches:
         print("No patches needed refresh")
         return False
+    
+    commit_msg = (
+        f"refresh patches against {commitish}\n\n"
+        f"patches: \n" + "\n".join(patches)
+    )
+    sh(f"git commit --no-verify -m '{commit_msg}' {' '.join(patches)}")
+    patch_texts = [p.replace("debian/patches/", "d/p/") for p in patches]
```
Comment (blackboxsw) [inline scripts/new_upstream_snapshot.py:355]: Yes funnily enough this was my first thought no this character for character before switching to an alternative proposal. +1

**Review by blackboxsw — APPROVED** [review]

**Pull request merged**